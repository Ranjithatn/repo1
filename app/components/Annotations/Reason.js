import React from "react";
import Select from "components/Select/Select"
import Label from "components/Label/Label";

export default function Reason({ formatMessage, onChange, selectedRegion, selectedReason, reasons }) {
  return (
    <div className="reason">
      {/* <div className="field is-horizontal">
        <div className="field-label is-normal">
          <Label text={(selectedRegion && formatMessage({id: selectedRegion})) || ""} />
        </div>
        <div className="field-body">
          <div className="field is-narrow">
            <div className="control"> */}
      <Label text={(selectedRegion && formatMessage({ id: selectedRegion })) || ""} />
      <Select
        disabled={!selectedRegion || !selectedRegion}
        className="is-fullwidth"
        formatMessage={formatMessage}
        onChange={onChange}
        value={selectedReason}
        defaultValue={formatMessage({ id: "Select Reason" })}
        options={reasons}
      />
      {/* </div>
          </div>
        </div>
      </div> */}
    </div>
  )
}
