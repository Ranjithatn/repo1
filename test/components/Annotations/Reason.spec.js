import React from "react";
import toJson from "enzyme-to-json";
import Reason from "../../../app/components/Annotations/Reason";
import Adapter from "enzyme-adapter-react-16";
import { configure, shallow, mount } from "enzyme";


configure({ adapter: new Adapter() });


xdescribe('<Reason />', () => {

  it("should render without crashing", () => {
    const component = mount(
      <Reason />
    );
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });
});