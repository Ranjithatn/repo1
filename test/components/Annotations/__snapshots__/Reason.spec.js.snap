// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Reason /> should render without crashing 1`] = `
<Reason>
  <div
    className="reason"
  >
    <Label
      text=""
    >
      <label
        className="label"
        htmlFor=""
        name=""
        style={Object {}}
      />
    </Label>
    <Select
      className="is-fullwidth"
      disabled={true}
    >
      <div
        className="select  is-fullwidth"
      >
        <select
          disabled={true}
          id=""
          name=""
          style={Object {}}
        />
      </div>
    </Select>
  </div>
</Reason>
`;
